src/qilib/data_set/data_set_io_reader.py

```diff
@@ -23,14 +23,19 @@
 
 
 class DataSetIOReader(ABC):
+    """ Abstract base class for data set io readers."""
+
+    def __init__(self) -> None:
+        """ This is an abstract base class and should not be instantiated directly."""
+        self._data_set: DataSet
 
-    @abstractmethod
     def bind_data_set(self, data_set: DataSet) -> None:
         """ Binds the DataSet to the DataSetIOReader.
 
         Args:
             data_set: The object that encompasses DataArrays.
         """
+        self._data_set = data_set
```

Comment:
(Remark) My inspector warns about this attribute being defined outside of the constructor. We can ignore this until we find a solution for this.

Reply:
✔️